jmh-benchmarks/src/main/java/org/apache/kafka/jmh/assignor/TargetAssignmentBuilderBenchmark.java

```diff
@@ -189,24 +159,21 @@ private Map<String, Assignment> generateMockInitialTargetAssignmentAndUpdateInve
         return initialTargetAssignment;
     }
 
-    private void createAssignmentSpec() {
-        Map<String, MemberSubscriptionAndAssignmentImpl> members = new HashMap<>();
+    private Map<String, ConsumerGroupMember> createMembers() {
+        return AssignorBenchmarkUtils.createHomogeneousMembers(
+            memberCount - 1,
+            this::memberId,
+            this::rackId,
+            allTopicNames
+        );
+    }
 
-        for (int i = 0; i < memberCount - 1; i++) {
-            String memberId = "member" + i;
+    private String memberId(int memberIndex) {
+        return "member" + memberIndex;
+    }
 
-            members.put(memberId, new MemberSubscriptionAndAssignmentImpl(
-                Optional.empty(),
-                Optional.empty(),
-                new TopicIds(new HashSet<>(allTopicNames), topicsImage),
-                Assignment.EMPTY
-            ));
-        }
-        groupSpec = new GroupSpecImpl(
-            members,
-            HOMOGENEOUS,
-            Collections.emptyMap()
-        );
+    private Optional<String> rackId(int memberIndex) {
+        return Optional.empty();
     }
```

Comment:
`createAssignmentSpec()` constructed the same subscription setup as `generateMockMembers()`.
This is now a shared call to `createMembers()` followed by `AssignorBenchmarkUtils.createGroupSpec()`.